### PR TITLE
Code Review Replace string values with companion object const values

### PR DESCRIPTION
**Pull request by Fouad-sys**
This pull request replaces all occurrences of database document string values by companion object const values as per the code review's request.

### CODE REVIEW

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt [hunk 1]
```diff
@@ -60,7 +62,7 @@ class AuthenticatedUser(
    * @param followed the [Profile] to follow.
    */
   suspend fun follow(followed: Profile) {
-    firestore.collection("users").document(followed.uid).update {
+    firestore.collection(ProfileDocument.Collection).document(followed.uid).update {
       arrayUnion("followers", user.uid)
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt:66]: Missing field.

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt [hunk 2]
```diff
@@ -72,7 +74,7 @@ class AuthenticatedUser(
    * @param unfollowed the [Profile] to unfollow.
    */
   suspend fun unfollow(unfollowed: Profile) {
-    firestore.collection("users").document(unfollowed.uid).update {
+    firestore.collection(ProfileDocument.Collection).document(unfollowed.uid).update {
       arrayRemove("followers", user.uid)
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt:78]: Missing field.

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/tournaments/StoreDocumentTournament.kt [hunk 3]
```diff
@@ -129,7 +133,7 @@ class StoreDocumentTournament(
       val round = document.stage?.toIntOrNull() ?: 1
       val results =
           store
-              .collection("games")
+              .collection(ChessDocument.Collection)
               .whereEquals("tournamentId", reference.uid)
               .whereEquals("roundDepth", round)
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/tournaments/StoreDocumentTournament.kt:138]: Missing field.

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/tournaments/StoreDocumentTournament.kt [hunk 4]
```diff
@@ -115,7 +119,7 @@ class StoreDocumentTournament(
     runCatching {
       val results =
           store
-              .collection("games")
+              .collection(ChessDocument.Collection)
               .whereEquals("tournamentId", reference.uid)
               .whereNotEquals("poolId", null)
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/tournaments/StoreDocumentTournament.kt:124]: Missing field.

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt [hunk 5]
```diff
@@ -99,7 +101,7 @@ class AuthenticatedUser(
    */
   val following: Flow<List<Profile>> =
       firestore
-          .collection("users")
+          .collection(ProfileDocument.Collection)
           .whereArrayContains("followers", user.uid)
           .orderBy("name")
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt:106]: Missing fields.

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt [hunk 6]
```diff
@@ -83,7 +85,7 @@ class AuthenticatedUser(
    * @param puzzle the [Puzzle] to mark as solved.
    */
   suspend fun solvePuzzle(puzzle: Puzzle) {
-    firestore.collection("users").document(this.uid).update {
+    firestore.collection(ProfileDocument.Collection).document(this.uid).update {
       arrayUnion("solvedPuzzles", puzzle.uid)
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt:89]: Missing field.

**Review by alexandrepiveteau — COMMENTED** [review]
Looks good to me ! A few fields may still be updated but otherwise nice work

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt (hunk 1 above)
Comment (Fouad-sys) [reply mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt]: Added in 715a39794e4fd9228eaaf11ad0bf8a1682f5edfd

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt (hunk 2 above)
Comment (Fouad-sys) [reply mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt]: Added in 715a39794e4fd9228eaaf11ad0bf8a1682f5edfd

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/tournaments/StoreDocumentTournament.kt (hunk 3 above)
Comment (Fouad-sys) [reply mobile/src/main/java/ch/epfl/sdp/mobile/application/tournaments/StoreDocumentTournament.kt]: Added in 715a39794e4fd9228eaaf11ad0bf8a1682f5edfd

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/tournaments/StoreDocumentTournament.kt (hunk 4 above)
Comment (Fouad-sys) [reply mobile/src/main/java/ch/epfl/sdp/mobile/application/tournaments/StoreDocumentTournament.kt]: Added in 715a39794e4fd9228eaaf11ad0bf8a1682f5edfd

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt (hunk 5 above)
Comment (Fouad-sys) [reply mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt]: Added in 715a39794e4fd9228eaaf11ad0bf8a1682f5edfd

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt (hunk 6 above)
Comment (Fouad-sys) [reply mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt]: Added in 715a39794e4fd9228eaaf11ad0bf8a1682f5edfd

**Review by matt989253 — COMMENTED** [review]
I found a few missing strings that were not refactored:
- "user", `StatefulTournamentDetailsScreenTest`, line 314
- "user", `StatefulTournamentDetailsScreenTest`, line 352

It seems that all the other were refactored (from a global string search through the project).

**Review by matt989253 — APPROVED** [review]

**Review by KurohanaJuri — APPROVED** [review]

**Pull request merged**